indico/modules/search/client/js/components/SideBar.jsx

```diff
@@ -6,130 +6,84 @@
 // LICENSE file for more details.
 
 import PropTypes from 'prop-types';
-import React, {useState} from 'react';
+import React from 'react';
 import {Table, Checkbox} from 'semantic-ui-react';
-import {useQueryParam, StringParam} from 'use-query-params';
 
 import './SideBar.module.scss';
 
-export default function SideBar({filterType}) {
-  const [speakerValue, setSpeakerValue] = useState('');
-  const [affiliationValue, setAffiliationValue] = useState('');
-
-  const [speakerFilter, setSpeakerFilter] = useQueryParam('speaker', StringParam);
-  const [affiliationFilter, setAffiliationFilter] = useQueryParam('affiliation', StringParam);
-
-  const addSearchFilter = (value, type) => {
-    if (type === 'speaker') {
-      setSpeakerFilter(speakerValue === value ? '' : value, 'pushIn');
-      setSpeakerValue(speakerValue === value ? '' : value);
-    } else if (type === 'affiliation') {
-      setAffiliationFilter(affiliationValue === value ? '' : value, 'pushIn');
-      setAffiliationValue(affiliationValue === value ? '' : value);
-    }
-  };
-
-  const handleChange = (e, {value}) => {
-    addSearchFilter(value.name, value.type);
+export default function SideBar({query, aggregations, onChange}) {
+  const handleChange = (type, value) => {
+    onChange && onChange(type in query && query[type] === value ? undefined : value, type);
```

Comment:
if (onChange) { ... }

Reply:
`type in query && ` isn't needed, `query[type]` will just be undefined otherwise

Reply:
Right! I don't think initially I was expecting the query to be an empty object.

Reply:
As for the `onChange`, I assume you're not a fan of the one-liner? As I could get rid of the brackets instead.

Reply:
Exactly, I don't mind it when using it in an expression or jsx attribute, but not in a normal statement